### PR TITLE
Cookiecutter update

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [x] (If applicable) Documentation has been added / updated (for bug fixes / features).
- [ ] (If applicable) Tests have been added.
- [x] This PR does not seem to break the templates.
- [x] CHANGELOG.rst has been updated (with summary of main changes).
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.

### What kind of change does this PR introduce?

* Updates the cookiecutter template to the latest commit
* CI now uses a `CI` folder to manage pip dependencies
* checks for `YAML` format and code quality are stricter (via `ruff`)
* Python3.13 has been added to the list of supported Python versions
* Development dependency and documentation updates
* `flake8-alphabetize` has been replaced by `ruff` 
* `numpydoc-validate` has been staged for integration in linting
* There is now a license statement in the top-level `__init__.py`

### Does this PR introduce a breaking change?

Not really. The linting checks are more intense.

### Other information:

`numpydoc-validate` is not enabled since the changes are significant and require input from multiple developers. These should come about in another PR.

### CODE REVIEW

File: src/xscen/catalog.py [hunk 1]
```diff
@@ -1018,7 +1021,7 @@ def unstack_id(df: Union[pd.DataFrame, ProjectCatalog, DataCatalog]) -> dict:
             [
                 col
                 for col in subset.columns
-                if bool(re.search(f"((_)|(^)){str(subset[col].iloc[0])}((_)|($))", ids))
+                if bool(re.search(f"((_)|(^)){subset[col].iloc[0]!s}((_)|($))", ids))
```
Comment (RondeauG) [inline src/xscen/catalog.py:1024]: Is that doing the right thing?

File: tox.ini [hunk 2]
```diff
@@ -1,31 +1,37 @@
 [tox]
-min_version = 4.0
+min_version = 4.18.0
 envlist =
     lint
-    py{39,310,311,312}
+    py{310,311,312,313}
```
Comment (RondeauG) [inline tox.ini:5]: Same.

File: tox.ini [hunk 3]
```diff
@@ -1,31 +1,37 @@
...
 opts =
     --colored
     --verbose
 
+[gh]
+python =
+    3.10 = py310-coveralls
+    3.11 = py311-coveralls
+    3.12 = py312-esmpy-coveralls
+    3.13 = py313-esmpy-coveralls
```
Comment (RondeauG) [inline tox.ini:19]: Same.

**Review by RondeauG — APPROVED** [review]

File: pyproject.toml [hunk 4]
```diff
@@ -173,7 +210,7 @@ py_version = 39
 
 [tool.mypy]
 files = "."
-python_version = 3.9
+python_version = 3.8
```
[the remark this answers is not in the document]
Comment (Zeitsperre) [reply pyproject.toml:213]: Ah, that's automatic, however the `mypy` checks aren't enforced at the moment. I'll need to update these once we decide to adopt them. Will reset to Python3.9.

File: src/xscen/catalog.py (hunk 1 above)
Comment (Zeitsperre) [reply src/xscen/catalog.py:1024]: Yes, this is proper using in f-strings.

See: https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals

**Pull request merged**